pkg/client/request.go

```diff
@@ -478,6 +483,27 @@ func (r *Request) Do() Result {
 			continue
 		}
 
+		if resp.StatusCode == http.StatusServiceUnavailable {
+			if retries == 10 {
+				// Give up after about 10 failed attempts.
+				return Result{respBody, created, err, r.codec}
```

Comment:
I thought about that and my feeling was that the extra continue/breaks needed would make the code harder to understand -- but I also worry about the repeated code. I shall refactor.